proto/documents/v1alpha/web_publishing.proto

```diff
@@ -0,0 +1,330 @@
+
+  // Removes site configuration from the local app backend.
+  rpc RemoveSite(RemoveSiteRequest) returns (google.protobuf.Empty);
+
+  // Lists configured sites.
+  rpc ListSites(ListSitesRequest) returns (ListSitesResponse);
+
+  // Lists web publication records for a given document among the configured web sites.
+  rpc ListWebPublicationRecords(ListWebPublicationRecordsRequest) returns (ListWebPublicationRecordsResponse);
+}
+
+// Request to add a site.
+message AddSiteRequest {
+  // Required. Site hostname.
+  string hostname = 1;
+
+  // Optional. Invite token for the site. Not needed
+  // if the site already knows our Account ID.
+  string invite_token = 2;
+}
+
+// Request to remove a site from the local app backend.
+message RemoveSiteRequest {
+  // Required. Site hostname.
+  string hostname = 1;
+}
+
+// Request to list configured sites.
+message ListSitesRequest {
+  // Optional. Number of items per page.
+  int32 page_size = 1;
+
+  // Optional. Token for a specific page.
+  string page_token = 2;
+}
+
+// Response with a list of sites.
+message ListSitesResponse {
+  // List of sites.
+  repeated SiteConfig sites = 1;
+
+  // Token for the next page if any.
+  string next_page_token = 2;
+}
+
+// Request to a list publication records for a given document
+// among the configured web sites.
+message ListWebPublicationRecordsRequest {
+  // Required. Document ID.
+  string document_id = 1;
+
+  // Optional. Specific version of a document.
+  string version = 2;
+}
+
+// Response with a list of publication records among the web sites.
+message ListWebPublicationRecordsResponse {
+  repeated WebPublicationRecord publications = 1;
+}
+
+// Local site configuration.
+message SiteConfig {
+  // Hostname of the site.
+  string hostname = 1;
+
+  // Our role on this site.
+  Member.Role role = 2;
+}
+
+// Member of the web site.
+message Member {
+  // Roles a member of the Mintter Site can have.
+  enum Role {
+    // Invalid default value.
+    ROLE_UNSPECIFIED = 0;
+
+    // Currently there can only be one owner of the site.
+    OWNER = 1;
+
+    // Editors are allowed to push content to the site.
+    EDITOR = 2;
+  }
+
+  // Mintter Account ID of the member.
+  string account_id = 1;
+
+  // Member's role on the site.
+  Role role = 2;
+}
+
+// Information about a Mintter Document published on a Mintter Site.
+// One Document ID can be published with different versions and/or
+// under different paths on a Mintter Web Site, so it can have
+// multiple of these records per site.
+message WebPublicationRecord {
+  string document_id = 1;
+
+  string version = 2;
+
+  string hostname = 3;
+
+  string path = 4;
+}
+
+// API exposed by the remote server that allows
+// publishing Mintter Documents to serve them on the Web.
+// The server implementing this API is responsible for authenticating
+// callers for its administrative RPCs.
+// This API is also exposed by the local application backend, which would act
```

Comment:
Since `WebSite` service will be exposed both locally in the local node (so the local node can proxy calls to the remote node if a header is provided) and via p2p by the remote server (so they can be called remotely), what if we fallback to call those functions locally only without proxying in the case the caller does not provide a header? In the case of the local node, this would not make sense and the node should check first if it's a site or not and return an error if its not. But on the remote site, however, if the NextJS wants to call one of those calls, like for example [GetPath](https://discord.com/channels/631404144640393217/1062451532483788801/1067621488280211587) ?

Reply:
I think you asked something similar in the other PR, but I don't remember what I answered (if anything).

The WebSite services exposed on the remote server doesn't need to require any headers, because this server will only be able to talk about its own web site, so nothing to specify in the header. No matter on what transport this service is exposed.

And the local backend which exposes this API for the purposes of proxying can't do anything meaningful without the header, because it won't know which hostname to proxy the requests to.